ecl/hql/hqlutil.cpp

```diff
@@ -397,11 +395,28 @@ static IHqlExpression * mapIfBlockRecord(HqlMapTransformer & mapper, IHqlExpress
 }
 
 
+static IHqlExpression * mapIfBlockCondition(HqlMapTransformer & mapper, IHqlExpression * expr)
+{
+    if (expr->getOperator() == no_select)
+    {
+        OwnedHqlExpr mapped = mapper.transformRoot(expr);
+        //Need to apply biases to fields that are keyed and ensure they are correctly cast
+        if (mapped->queryType() != expr->queryType())
```

Comment:
If block conditions were not correctly modified when the fields were mapped to the real index fields.  This meant the size of a raw index record containing ifblocks was wrong.